src/variety/cbblock.js

```diff
@@ -201,14 +301,42 @@ FileIO:{
 	}
 },
 
+"FileIO@dbchoco":{
+	decodeData : function(){
+		this.decodeCell( function(cell,ca){
+			if(ca.charAt(0)==="-"){cell.ques=6; ca = ca.substr(1);}
+
+			if(ca==="0") {cell.qnum = -2;}
+			else if(ca!=="." && (+ca)>0){cell.qnum = +ca;}
+		});
+		this.decodeBorderAns();
+	},
+	encodeData : function(){
+		this.encodeCell(function(cell){
+			var ca = "";
+			if(cell.ques===6) { ca += "-";}
+
+			if (cell.qnum===-2) { ca += "0";}
+			else if (cell.qnum!==-1){ ca += cell.qnum.toString();}
+
+			if(ca === "") { ca = "."; }
+			return ca+" ";
+		});
+		this.encodeBorderAns();
+	}
+},
+
 //---------------------------------------------------------
 // 正解判定処理実行部
 AnsCheck:{
 	checklist : [
 		"checkSingleBlock",
-		"checkBlockNotRect",
-		"checkDifferentShapeBlock",
-		"checkLargeBlock"
+		"checkSmallNumberArea@dbchoco",
+		"checkBlockNotRect@cbblock",
+		"checkDifferentShapeBlock@cbblock",
+		"checkEqualShapes@dbchoco",
+		"checkLargeBlock",
+		"checkLargeNumberArea@dbchoco",
```

Comment:
- I felt the order was a bit strange here when playing; particularly, if there's an area with two shapes which have different shapes, and an area with three shapes, then the error message that's displayed is that the first area has different shapes, while it seems the other area is fine. Seems number of areas should be checked first: <img width="544" alt="Screenshot 2019-05-07 at 10 58 26" src="https://user-images.githubusercontent.com/132113/57290042-ea628b00-70bc-11e9-9866-d16e9c397d68.png">
- we should check for dead-end lines, these are currently ignored